### ISSUE TITLE
Docker Container - Tensorboard logging fails

### ISSUE THREAD

**Issue by DBadLuck** [ultralytics/yolov5#2573]
## ❔Question
How can I start Tensorboard logging in Docker and how is it visable outside docker on the localhost adress?

## Additional context
I start the Docker Container with the information given in your Docker discription:
sudo docker run --ipc=host --gpus all -it -p 8888:6006

With the additional portmaping 6006 -> 8888 I should see the tensorboard outside of Docker on my local machine on the web adresse "localhost:8888" but I didn't see anything.

**Comment by glenn-jocher** [comment]
@DBadLuck Locally pointing your browser to http://localhost:6006/ will work (once tensorboard is started and is logging), and the portmapping you have there should work also. 

One important point is that tensorboard logging is not automatically start when train.py starts, it must be started manually, i.e. by running this command from the yolov5/ directory:
```
tensorboard --logdir runs/train
```

**Comment by glenn-jocher** [comment]
You can also run it in the background so it doesn't utilize a console tab:
```bash
tensorboard --logdir runs/train &
python train.py ...
```

**Comment by Ownmarc** [comment]
Thats gave me an idea.. 

@glenn-jocher would you be against adding a possible arg --tensorbaord to train.py so it can be started with the python train command ?

**Comment by DBadLuck** [comment]
> @DBadLuck Locally pointing your browser to http://localhost:6006/ will work (once tensorboard is started and is logging), and the portmapping you have there should work also.
> 
> One important point is that tensorboard logging is not automatically start when train.py starts, it must be started manually, i.e. by running this command from the yolov5/ directory:
> 
> ```
> tensorboard --logdir runs/train
> ```

@glenn-jocher Thanks for your quick reply.

After typing in your command 
`tensorboard --logdir runs/train`

the Comandline of the Docker image replies:
`# tensorboard --logdir runs/train
TensorFlow installation not found - running with reduced feature set.
Traceback (most recent call last):
  File "/opt/conda/bin/tensorboard", line 8, in <module>
    sys.exit(run_main())
  File "/opt/conda/lib/python3.8/site-packages/tensorboard/main.py", line 75, in run_main
    app.run(tensorboard.main, flags_parser=tensorboard.configure)
  File "/opt/conda/lib/python3.8/site-packages/absl/app.py", line 303, in run
    _run_main(main, args)
  File "/opt/conda/lib/python3.8/site-packages/absl/app.py", line 251, in _run_main
    sys.exit(main(argv))
  File "/opt/conda/lib/python3.8/site-packages/tensorboard/program.py", line 289, in main
    return runner(self.flags) or 0
  File "/opt/conda/lib/python3.8/site-packages/tensorboard/program.py", line 305, in _run_serve_subcommand
    server = self._make_server()
  File "/opt/conda/lib/python3.8/site-packages/tensorboard/program.py", line 414, in _make_server
    app = application.TensorBoardWSGIApp(
  File "/opt/conda/lib/python3.8/site-packages/tensorboard/backend/application.py", line 143, in TensorBoardWSGIApp
    return TensorBoardWSGI(
  File "/opt/conda/lib/python3.8/site-packages/tensorboard/backend/application.py", line 256, in __init__
    raise ValueError(
ValueError: Duplicate plugins for name projector
`

**Comment by glenn-jocher** [comment]
@DBadLuck well that's a new message, thanks for letting us know. I'll try to reproduce later today.

**Comment by mattpopovich** [comment]
I can reproduce, I get the same error when running in the docker container (found this a week or so ago but forgot to open an issue):
```
root@PC:/home/user/git/yolov5# which tensorboard
/opt/conda/bin/tensorboard
root@PC:/home/user/git/yolov5# tensorboard --logdir runs
TensorFlow installation not found - running with reduced feature set.
Traceback (most recent call last):
  File "/opt/conda/bin/tensorboard", line 8, in <module>
    sys.exit(run_main())
  File "/opt/conda/lib/python3.8/site-packages/tensorboard/main.py", line 75, in run_main
    app.run(tensorboard.main, flags_parser=tensorboard.configure)
  File "/opt/conda/lib/python3.8/site-packages/absl/app.py", line 303, in run
    _run_main(main, args)
  File "/opt/conda/lib/python3.8/site-packages/absl/app.py", line 251, in _run_main
    sys.exit(main(argv))
  File "/opt/conda/lib/python3.8/site-packages/tensorboard/program.py", line 289, in main
    return runner(self.flags) or 0
  File "/opt/conda/lib/python3.8/site-packages/tensorboard/program.py", line 305, in _run_serve_subcommand
    server = self._make_server()
  File "/opt/conda/lib/python3.8/site-packages/tensorboard/program.py", line 414, in _make_server
    app = application.TensorBoardWSGIApp(
  File "/opt/conda/lib/python3.8/site-packages/tensorboard/backend/application.py", line 143, in TensorBoardWSGIApp
    return TensorBoardWSGI(
  File "/opt/conda/lib/python3.8/site-packages/tensorboard/backend/application.py", line 256, in __init__
    raise ValueError(
ValueError: Duplicate plugins for name projector
```
I looked into it briefly but didn't find a solution. Was easier to just run tensorboard from outside the container.

**Comment by glenn-jocher** [comment]
I can reproduce also. Found a PyTorch issue on the topic here: https://github.com/pytorch/pytorch/issues/22676

Tried a few solutions but did not have any success. If anyone finds a fix please update us.

**Comment by glenn-jocher** [comment]
I opened PR #2576 to update tensorboard version requirement to the latest. Probably won't solve this issue but it couldn't hurt.

**Comment by feaudev** [comment]
Answer 1. 
%tensorboard --logdir runs/train --port 6006 --host 0.0.0.0  # needs '--host 0.0.0.0' inside docker container 

Answer 2.
// cannot coexist with tensorboard-2.4.1, so should be removed before installing tensorbaord-2.4.1
// from tensorboard import default
// print(default.get_dynamic_plugins()) # to see multiple entries of tensorboard.plugins.projector.projector_plugin.ProjectorPlugin'
RUN pip uninstall -y nvidia-tensorboard-plugin-dlprof
RUN pip uninstall -y nvidia-tensorboard

**Comment by glenn-jocher** [comment]
@feaudev thanks! Are you suggesting the cause of the problem is a conflict with tensorboard and nvidia-tensorboard, and that the solution is to add the `RUN pip uninstall nvidia-tensorboard` lines to the Dockerfile?

**Comment by feaudev** [comment]
@glenn-jocher yes.

**Comment by glenn-jocher** [comment]
@feaudev could you take a look at PR #2622 to see if this implements your suggestion correctly?

**Comment by feaudev** [comment]
@glenn-jocher l left an approval review.

**Comment by glenn-jocher** [comment]
@DBadLuck @Ownmarc @mattpopovich tensorboard logging in Docker issue should be resolved now in PR #2622. Credit to @feaudev for the idea. Please git pull or reclone to receive this update and let us know if you run into any other problems!
